src/coreclr/vm/amd64/AsmMacros.inc

```diff
@@ -206,6 +206,26 @@ INLINE_GETTHREAD macro Reg
 
         endm
 
+;
+; Inlined macro to get the current thread's allocation context
+; Trashes rax and r11
+;
+
+INLINE_GET_ALLOC_CONTEXT macro Reg
+
+        EXTERN _tls_index: DWORD
+        EXTERN t_thread_alloc_context: DWORD
+
+        mov     r11d, [_tls_index]
+        mov     rax, gs:[OFFSET__TEB__ThreadLocalStoragePointer]
+        mov     rax, [rax + r11 * 8]
+        mov     r11d, SECTIONREL t_thread_alloc_context
+        add     rax, r11
+        mov     Reg, rax
```

Comment:
```suggestion
        mov     Reg, SECTIONREL t_thread_alloc_context
        add     Reg, rax
```
Save extra instruction?

Reply:
Looks like I need to undo this because of instruction size requirements (SECTIONREL needs to be moved into a dword-sized register alias).